scripts/create_test_subset.py

```diff
@@ -401,8 +491,16 @@ def transfer_analyses(
                             project,
                             (str(sg['id']), new_sg_map[s['externalId']][0]),
                         ),
-                        status=AnalysisStatus(analysis['status'].lower().replace('_', '-')),
-                        sequencing_group_ids=new_sg_map[s['externalId']],
+                        status=AnalysisStatus(
+                            analysis['status'].lower().replace('_', '-')
+                        ),
+                        sequencing_group_ids=get_new_sg_id(
```

Comment:
Maybe we can 'fail faster' here. So call this function a little earlier, validate that we're getting a list of length one, and return an error otherwise.